superset/utils/urls.py

```diff
@@ -33,3 +33,12 @@ def headless_url(path: str, user_friendly: bool = False) -> str:
 def get_url_path(view: str, user_friendly: bool = False, **kwargs: Any) -> str:
     with current_app.test_request_context():
         return headless_url(url_for(view, **kwargs), user_friendly=user_friendly)
+
+
+def get_screenshot_explorelink(url: Optional[str]) -> Optional[str]:
```

Comment:
Can you add a docstring to the function explaining what it does? As for types, it's better to have the signature be `(url: str) -> str:`, otherwise you need to handle the case where `url` is `None`.

One thing I was thinking is that since this function is currently very specific we could make it a bit more generic, so it has utility outside of your use case. Eg:

```python
def modify_url_query(url: str, **kwargs: Any) -> str:
    """
    Replace or add parameters to a URL.
    """
    parts = list(urllib.parse.urlsplit(url))
    params = urllib.parse.parse_qs(parts[3])
    for k, v in kwargs.items():
        if not isinstance(v, list):
            v = [v]
        params[k] = v
    
    parts[3] = "&".join(f"{k}={urllib.parse.quote(v[0])}" for k, v in params.items())
    return urllib.parse.urlunsplit(parts)
```

Then you can call it for your use case as:

```python
url = modify_url_query(self._content.url, standalone='0')
```

Reply:
ok I think that is a cool generic function, thank you! When I had `(url: string) -> string` I got a type error, and since i was rushing against the cherry deadline I made it optional, will change back and think it through. Thank you!

Reply:
If you had an error with `(url: str) -> str` it means that in some place the function was potentially being called with a non-string value (probably `None`, since you fixed it by adding `Optional`). Since your original function did not handle the case when the argument was not a string, this means that at some point the code would break.

In other words, if your function has a signature `(url: Optional[str])` it should be able to handle being passed `url=None`. But more importantly, the types in the signature should make sense — would we ever want to pass `None` as a URL to a function that changes parameters in the URL?

The proper way to fix the type checking is not relaxing the signature, but adding guards to the code:

```python
if url is not None:
    explore_url = get_screenshot_explorelink(url)
```

Reply:
I was thinking of adding an exception into the modify_url_query function, but this feels like it would look better in the email.

Reply:
Since we do end up using the URL in the email later I was thinking that we could do this:
```
        url = ""
        if self._content.url is not None:
            url = modify_url_query(self._content.url, standalone="0")
```